libraries/drivers_appserver_base.rb

```diff
@@ -37,18 +40,30 @@ def appserver_config
         raise NotImplementedError
       end
 
+      # Creates a monit config file for managing the appserver and then notifies
+      # monit to reload it.
       def add_appserver_monit
-        opts = { app_shortname: app['shortname'], appserver_name: adapter, appserver_command: appserver_command,
-                 deploy_to: deploy_dir(app), environment: environment }
-
-        file_path = File.join(node['monit']['basedir'], "#{opts[:appserver_name]}_#{opts[:app_shortname]}.monitrc")
+        opts = {
+          app_shortname: app['shortname'],
+          adapter: adapter,
+          appserver_command: appserver_command,
+          appserver_name: adapter,
+          deploy_to: deploy_dir(app),
+          environment: environment,
+          source_cookbook: appserver_monit_template_cookbook
+        }
+        file_path = File.join(node['monit']['basedir'],
+                              "#{opts[:appserver_name]}_#{opts[:app_shortname]}.monitrc")
         context.template file_path do
           mode '0640'
-          source 'appserver.monitrc.erb'
+          source "#{opts[:adapter]}.monitrc.erb"
+          cookbook opts[:source_cookbook].to_s
```

Comment:
This should let a cookbook override the template if a user needs complete control. The worker driver already does this.